portfolio/src/main/java/com/google/sps/servlets/BusinessCardServlet.java

```diff
@@ -0,0 +1,186 @@
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * This class will handle the storage/retrieval of information from the
+ * Business Card Drop on the portfolio homepage.
+ */
+@WebServlet("/biz-card")
+public class BusinessCardServlet extends HttpServletWithUtilities {
+
+  // Store the BlobstoreService instance for the application. Same instance as other files.
+  private final BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+  // Store the Datastore Service instance. Will hold all BizCard entities.
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  // Store the ImagesService instance which will be used to get download URLs for the images
+  // in blobstore.
+  private final ImagesService imagesService = ImagesServiceFactory.getImagesService();
+
+  /**
+   * Add a business card to the Datastore.
+   * DO NOT ACCESS DIRECTLY - should only be accessed by Blobstore.
+   * Point Client to the blobstore upload URL, which can be retrieved from the BlobstoreUrlServlet
+   *
+   * Image must have been uploaded inside a form element with the name "bizCard".
+   * If this changes, change the formElementName parameter in the getImageBlobKey function.
+   *
+   * @param request HTTP request sent from client for POST request
+   * @param response HTTP response to be sent to client
+   * @throws IOException if an IO error occurs while the request is being processed by the servlet.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the blobKey of the image so it can be located in the blobstore.
+    BlobKey bizCardBlobKey = getImageBlobKey(request, "bizCard");
+
+    // If a blob key was not generated, an error occurred with the file.
+    if(bizCardBlobKey == null) {
+      response
+          .sendError(400, "No image uploaded or file was invalid (corrupted, not image, etc.)");
+      return;
+    }
+
+    // Get URL to download image
+    String bizCardURL = getImageUrl(bizCardBlobKey);
+
+    // Add Business Card URL to the datastore
+    Entity businessCardEntity = new Entity("BizCard");
+    businessCardEntity.setProperty("bizCard", bizCardURL);
+    datastore.put(businessCardEntity);
```

Comment:
Looks like datastore.put() could have three errors: 1) Illegal Argument Error, 2) Concurrent Modification Error, and 3) Datastore Service error. In cases 1) and 2), something catastrophic happened that cannot be recovered from, so a generic 500 error is probably fine. These errors should not happen, since the entity is brand new (so how can it already be concurrently modified) and the argument is not illegal since I just made it. 

In case 3), the error should be passed to the user and program should recover gracefully.

Added handling for this and created an issue to handle other instances of this throughout the code #30